OrbitGl/Batcher.cpp

```diff
@@ -95,4 +114,86 @@ void Batcher::GetBoxGradientColors(Color color, Color* colors) {
 void Batcher::Reset() {
   line_buffer_.Reset();
   box_buffer_.Reset();
+  triangle_buffer_.Reset();
+}
+
+//----------------------------------------------------------------------------
+void Batcher::Draw(bool a_Picking) {
+  glPushAttrib(GL_ENABLE_BIT | GL_COLOR_BUFFER_BIT);
+  glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+  glDisable(GL_CULL_FACE);
+  glEnableClientState(GL_VERTEX_ARRAY);
+  glEnableClientState(GL_COLOR_ARRAY);
+  glEnable(GL_TEXTURE_2D);
+
+  DrawBoxBuffer(a_Picking);
+  DrawLineBuffer(a_Picking);
+  DrawTriangleBuffer(a_Picking);
+
+  glDisableClientState(GL_COLOR_ARRAY);
+  glDisableClientState(GL_VERTEX_ARRAY);
+  glPopAttrib();
+}
+
+//----------------------------------------------------------------------------
+void Batcher::DrawBoxBuffer(bool a_Picking) {
+  Block<Box, BoxBuffer::NUM_BOXES_PER_BLOCK>* boxBlock =
```

Comment:
There are iterators in the BlockChain that might make this a bit more readable.

Reply:
The iterators in BlockChain provide iterating over all elements in the blockchain, not iterating over blocks (which is what I need here). 

Btw, this code was moved from a different location and not really added in this PR. That's also why the naming is not according to our style guide. I'll change this here.